packages/react/src/template.tsx

```diff
@@ -137,56 +160,47 @@ export function BaseTemplate({
 
   function renderProp(
     propKey: string,
-    stateNode: JSONValue,
-    parentNode: ComponentDataState,
-    path: string
+    propValue: JSONValue,
+    parentNode: ComponentDataState | ReferencedComponentDataState,
+    path: Path
   ): any {
-    if (
-      isEventDataState(stateNode) ||
-      (Array.isArray(stateNode) && stateNode.length > 0 && stateNode.every(isEventDataState))
-    ) {
-      return (payload: any) => {
-        const nodes = Array.isArray(stateNode) ? stateNode : [stateNode]
-        for (const node of nodes) {
-          // React events (e.g. from onPress) contain cyclic structures that can't be serialized
-          // with JSON.stringify and also provide little to no value for the server.
-          // tbd: figure a better way to handle this in a cross-platform way
-          if (payload?.nativeEvent) {
-            payload = '[native code]'
-          }
-          // tbd: in the future, this should resolve with whatever server responded as a result of the event
-          // this is going to be more efficient and universal than the current approach
-          onEvent?.({
-            target: { key: parentNode.key, path, component: parentNode.component },
-            name: propKey,
-            action: node.action,
-            payload,
-          })
+    if (isEventDataState(propValue)) {
+      return async (payload: any) => {
+        // React events (e.g. from onPress) contain cyclic structures that can't be serialized
+        // with JSON.stringify and also provide little to no value for the server.
+        // tbd: figure a better way to handle this in a cross-platform way
+        if (payload?.nativeEvent) {
+          payload = '[native code]'
```

Comment:
hmm yeah this isn't ideal. But we can figure out a way to improve it later

Reply:
at least we're temporarily as fancy as dev tools console.log 😆